spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/stores/FormStore.java

```diff
@@ -166,4 +151,67 @@ private void addLayerByConfig(SCFormConfig config) {
             hasForms.onNext(storeForms.size() > 0);
         }
     }
+
+    @Override
+    public Observable<SCSpatialFeature> unSynced() {
+        //TODO get all unsynced features based on upload flag
+        return null;
+    }
+
+    @Override
+    public void upload(final SCSpatialFeature scSpatialFeature) {
+
+        final SpatialConnect sc = SpatialConnect.getInstance();
+        SCSensorService sensorService = sc.getSensorService();
+        sensorService.isConnected().subscribe(new Action1<Boolean>() {
+            @Override
+            public void call(Boolean connected) {
+                if (connected) {
+                    //make sure backendService is running
+                    sc.serviceRunning(SCBackendService.serviceId())
+                            .filter(new Func1<SCServiceStatusEvent, Boolean>() {
+                                @Override
+                                public Boolean call(SCServiceStatusEvent scServiceStatusEvent) {
+                                    return scServiceStatusEvent.getStatus()
+                                            .equals(SCServiceStatus.SC_SERVICE_RUNNING);
+                                }
+                            })
+                            .subscribe(new Action1<SCServiceStatusEvent>() {
+                                @Override
+                                public void call(SCServiceStatusEvent scServiceStatusEvent) {
+                                    SCBackendService backendService = sc.getBackendService();
+                                    Integer formId;
+                                    SCFormConfig c = storeForms.get(scSpatialFeature.getKey().getLayerId());
+                                    formId = Integer.parseInt(c.getId());
+                                    if (formId != null) {
+                                        HashMap<String, Object> formSubmissionPayload = new HashMap<>();
+                                        formSubmissionPayload.put("form_id", formId);
+                                        formSubmissionPayload.put("feature", scSpatialFeature);
+                                        try {
+                                            String payload = SCObjectMapper.getMapper().writeValueAsString(formSubmissionPayload);
+                                            SCMessageOuterClass.SCMessage message = SCMessageOuterClass.SCMessage.newBuilder()
+                                                    .setAction(SCCommand.DATASERVICE_CREATEFEATURE.value())
+                                                    .setPayload(payload)
+                                                    .build();
+
+                                            backendService.publishReplyTo("/store/form", message)
```

Comment:
iOS has a `syncChannel` method. That may be easier and this can probably go up into the parent class someday if its using instance methods to accomplish the upload.